File: server/router/miser/miser_category.go
package miser

import (
	"github.com/gin-gonic/gin"
	"github.com/springbear2020/self-hub/server/middleware"
)

type MiserCategoryRouter struct{}

func (s *MiserCategoryRouter) InitMiserCategoryRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	miserCategoryRouter := Router.Group("miserCategory").Use(middleware.OperationRecord())
	{
		miserCategoryRouter.POST("createMiserCategory", miserCategoryApi.CreateMiserCategory)
		miserCategoryRouter.DELETE("deleteMiserCategory", miserCategoryApi.DeleteMiserCategory)
		miserCategoryRouter.DELETE("deleteMiserCategoryByIds", miserCategoryApi.DeleteMiserCategoryByIds)
		miserCategoryRouter.PUT("updateMiserCategory", miserCategoryApi.UpdateMiserCategory)
	}

	miserCategoryRouterWithoutRecord := Router.Group("miserCategory")
	{
		miserCategoryRouterWithoutRecord.GET("findMiserCategory", miserCategoryApi.FindMiserCategory)
		miserCategoryRouterWithoutRecord.GET("getMiserCategoryList", miserCategoryApi.GetMiserCategoryList)
		miserCategoryRouterWithoutRecord.GET("listMiserCategoryList", miserCategoryApi.ListMiserCategoryList)
	}

	miserCategoryRouterWithoutAuth := PublicRouter.Group("miserCategory")
	{
		_ = miserCategoryRouterWithoutAuth
	}
}

File: web/src/api/miser/miser_category.js
import service from '@/utils/request'

export const createMiserCategory = (data) => {
  return service({
    url: '/miserCategory/createMiserCategory',
    method: 'post',
    data
  })
}

export const deleteMiserCategory = (params) => {
  return service({
    url: '/miserCategory/deleteMiserCategory',
    method: 'delete',
    params
  })
}

export const deleteMiserCategoryByIds = (params) => {
  return service({
    url: '/miserCategory/deleteMiserCategoryByIds',
    method: 'delete',
    params
  })
}

export const updateMiserCategory = (data) => {
  return service({
    url: '/miserCategory/updateMiserCategory',
    method: 'put',
    data
  })
}

export const findMiserCategory = (params) => {
  return service({
    url: '/miserCategory/findMiserCategory',
    method: 'get',
    params
  })
}

export const getMiserCategoryList = (params) => {
  return service({
    url: '/miserCategory/getMiserCategoryList',
    method: 'get',
    params
  })
}

export const listMiserCategoryList = () => {
  return service({
    url: '/miserCategory/listMiserCategoryList',
    method: 'get'
  })
}
